cmd/juju/commands/upgradecontroller.go

```diff
@@ -262,19 +272,13 @@ func initCAASVersions(
 	}, nil
 }
 
-func (c *upgradeControllerCommand) upgradeIAASController(ctx *cmd.Context) error {
+func (c *upgradeControllerCommand) upgradeIAASController(ctx *cmd.Context, controllerModel string) error {
 	jcmd := &upgradeJujuCommand{baseUpgradeCommand: baseUpgradeCommand{
 		upgradeMessage: "upgrade to this version by running\n    juju upgrade-controller",
 	}}
 	jcmd.SetClientStore(c.ClientStore())
 	wrapped := modelcmd.Wrap(jcmd)
-	controllerName, err := c.ControllerName()
-	if err != nil {
-		return errors.Trace(err)
-	}
-	fullControllerModelName := modelcmd.JoinModelName(controllerName,
```

Comment:
I think you still need this bit, which explicitly prepends the controller name.

Reply:
thanks, I've amended!